contracts/linear/src/validator_pool.rs

```diff
@@ -417,17 +412,39 @@ impl LiquidStakingContract {
     pub fn update_weight(&mut self, validator_id: AccountId, weight: u16) {
         self.assert_running();
         self.assert_manager();
-        self.validator_pool.update_weight(&validator_id, weight);
+        let old_weight = self.validator_pool.update_weight(&validator_id, weight);
+        Event::ValidatorUpdatedWeight {
+            account_id: &validator_id,
+            old_weight,
+            new_weight: weight,
+        }
+        .emit();
     }
 
     pub fn update_weights(&mut self, validator_ids: Vec<AccountId>, weights: Vec<u16>) {
         self.assert_running();
         self.assert_manager();
         require!(validator_ids.len() == weights.len(), ERR_BAD_VALIDATOR_LIST);
+
+        let mut account_ids = Vec::new();
+        let mut old_weights = Vec::new();
+        let mut new_weights = Vec::new();
+
         for i in 0..validator_ids.len() {
```

Comment:
btw shall we add a limitation of the max amount of validators that can be updated in one batch?

Reply:
Sounds good! This logic can save gas. Have just added a limit `MAX_UPDATE_WEIGHTS_COUNT`!